### ISSUE TITLE
documentation: create migration guide

### ISSUE THREAD

**Issue by x1B** [LaxarJS/laxar#420]
Let's create a helpful markdown migration guide (LaxarJS v1 to v2), which we can reference from the blog upon release.

**Comment by x1B** [comment]
@jpommerening @jonaswjs @alex3683 reviews appreciated

**Comment by jpommerening** [comment]
That was a great read!

- https://github.com/LaxarJS/laxar/blob/feature/420-migration-guide/docs/manuals/upgrade_guide.md#tooling-npm-and-webpack
  + "[…] are much smaller then grunt-laxar." *than
- https://github.com/LaxarJS/laxar/blob/feature/420-migration-guide/docs/manuals/upgrade_guide.md#json-schema-now-processed-at-build-time
  + Should we mention the expansion/interpolation of `${stuff}` here?

**Comment by alex3683** [comment]
- Perhaps mention, that changes listed in the changelog may already be reverted by a followup change (thus justifying the need for the upgrade guide).
- "we chose switch to NPM for front end artifacts" --> "we chose **to** switch to NPM for front end artifacts"?
- https://github.com/LaxarJS/laxar/blob/feature/420-migration-guide/docs/manuals/upgrade_guide.md#stylestemplates-for-widgetscontrols-lookup-by-name-only
  - Perhaps describe in a little more detail, how the name is used to resolve the artifacts? (Very small example perhaps)
- "Some widgets may use toUrl-method" --> "Some widgets may use **the** toUrl-method" ?

Overall 👍🏻
